subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/GradleBuildExperimentRunner.java

```diff
@@ -182,11 +186,23 @@ private File determineGradleUserHome(GradleInvocationSpec invocationSpec) {
     private GradleScenarioDefinition createScenarioDefinition(GradleBuildExperimentSpec experimentSpec, InvocationSettings invocationSettings, GradleInvocationSpec invocationSpec) {
         GradleDistribution gradleDistribution = invocationSpec.getGradleDistribution();
         List<String> cleanTasks = invocationSpec.getCleanTasks();
+        File gradlePropertiesFile = new File(invocationSettings.getProjectDir(), "gradle.properties");
+        Properties gradleProperties = new Properties();
+        try (InputStream inputStream = Files.newInputStream(gradlePropertiesFile.toPath())) {
+            gradleProperties.load(inputStream);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        String[] jvmOptsFromGradleProperties = gradleProperties.getProperty("org.gradle.jvmargs").split(" ");
```

Comment:
Is there any possibility that the jvm arg entry including spaces, like `-XX:HeapDumpPath=C:\Program Files\XXX` ?

Reply:
Yes, there is. Though that is not the case for the performance test projects we are using. Do you have an idea how we could solve this issue?

Reply:
Oh, this is only for internal performance testing, then no problem. I thought this is changing gradle distribution behavior.

Reply:
I changed the title to reflect that this is only for performance tests.